lib/puppet/resource/type.rb

```diff
@@ -413,6 +413,9 @@ def assign_defaults(resource, scope, resource_hash)
 
   def validate_resource_hash(resource, resource_hash)
     Puppet::Pops::Types::TypeMismatchDescriber.validate_parameters(resource.to_s, parameter_struct, resource_hash)
+  rescue Puppet::ParseError => e
```

Comment:
ok - as a dirty fix there needs to be a ticket to fix it the right way, and a comment in the code that explains the dirty fix and that mentions the ticket with the real fix.